internal/server/questions.go

```diff
@@ -223,6 +223,14 @@ func ginQuestionsHandler(s *Server) func(c *gin.Context) {
 			return
 		}
 
+		timeExtension := stepRun.EndedAt.Add(time.Minute * time.Duration(step.HitArgs.Timeout))
+		remainingTime := timeExtension.Sub(time.Now())
+		if stepRun.Status != stepRunModel.StatusRUNNING && remainingTime < 0 {
```

Comment:
I think same logic as answer.go. I think so, maybe not, let me know if there's any reason to have different logic on these.

Reply:
Yeah i agreed it should be the same logic as answers.go here.